src/Aspire.Hosting.Docker/EnvFile.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.Docker;
+
+internal sealed class EnvFile
+{
+    private readonly List<string> _lines = [];
+    private readonly HashSet<string> _keys = [];
+
+    public static EnvFile Load(string path)
+    {
+        var envFile = new EnvFile();
+        if (!File.Exists(path))
+        {
+            return envFile;
+        }
+
+        foreach (var line in File.ReadAllLines(path))
+        {
+            envFile._lines.Add(line);
```

Comment:
Instead of keeping track of all lines, why not just append to the file if there are lines to add?